react-router/lib/IndexRoute.d.ts

```diff
@@ -15,8 +15,8 @@ type ComponentsCallback = (err: any, components: RouteComponents) => any;
 export interface IndexRouteProps {
     component?: RouteComponent;
     components?: RouteComponents;
-    getComponent?(nextState: RouterState, callback: ComponentCallback): undefined;
```

Comment:
I do not think this is necessary. Changing `undefined` to `void` should be good enough